tests/linearizability/failpoints.go

```diff
@@ -115,44 +125,55 @@ const (
 )
 
 func (f goFailpoint) Trigger(t *testing.T, ctx context.Context, clus *e2e.EtcdProcessCluster) error {
-	var member e2e.EtcdProcess
-	switch f.target {
-	case AnyMember:
-		member = clus.Procs[rand.Int()%len(clus.Procs)]
-	case Leader:
-		member = clus.Procs[clus.WaitLeader(t)]
-	default:
-		panic("unknown target")
-	}
+	member := f.pickMember(t, clus)
 	address := fmt.Sprintf("127.0.0.1:%d", member.Config().GoFailPort)
-	err := setupGoFailpoint(address, f.failpoint, f.payload)
-	if err != nil {
-		return fmt.Errorf("gofailpoint setup failed: %w", err)
-	}
-	if f.trigger != nil {
-		err = f.trigger(ctx, member)
+
+	triggerCtx, cancel := context.WithTimeout(ctx, time.Second)
+	defer cancel()
+
+	// TODO: Validate that etcd crashed due to failpoint trigger.
+	for member.IsRunning() {
+		err := setupGoFailpoint(triggerCtx, address, f.failpoint, f.payload)
 		if err != nil {
-			return fmt.Errorf("triggering gofailpoint failed: %w", err)
+			t.Logf("gofailpoint setup failed: %v", err)
+		}
+		if f.trigger != nil {
+			err = f.trigger(triggerCtx, member)
+			if err != nil {
+				t.Logf("triggering gofailpoint failed: %v", err)
+			}
+		}
+		err = member.Wait(triggerCtx)
+		if err != nil && !strings.Contains(err.Error(), "unexpected exit code") {
+			return fmt.Errorf("failed to trigger a process panic within %s, err: %w", triggerTimeout, err)
 		}
 	}
```

Comment:
This isn't correct. A gofailpoint isn't necessarily causing etcd crash. It works for now because all current failpoints are `panic`, but we may add other failpoints, e.g. `sleep`.

Reply:
Yes but we currently only support crash. I have added a refactor PR that makes it explicit.